vulnerable_people_form/integrations/submission_tracing.py

```diff
@@ -0,0 +1,72 @@
+import json
+import time
+import hashlib
+import random
+
+from flask import request, current_app
+
+
+def client_ip():
+    if request:
+        if request.environ.get("HTTP_X_FORWARDED_FOR") is None:
+            return request.environ["REMOTE_ADDR"]
+        else:
+            # if behind a proxy
+            return request.environ["HTTP_X_FORWARDED_FOR"]
+    return None
+
+
+def user_agent():
+    result = {"platform": None, "browser": None, "version": None, "string": None}
+    if request:
+        ua_obj = request.user_agent
+        result["platform"] = ua_obj.platform
+        result["browser"] = ua_obj.browser
+        result["version"] = ua_obj.version
+        result["string"] = ua_obj.string
+    return result
+
+
+def peppering(obj):
+    if obj is not None and current_app:
+        pepper = current_app.config.get("SUBMISSION_TRACING_PEPPER")
+        if pepper is not None:
+            m = hashlib.sha256()
+            # convert to string and lower case
+            objstr = str(obj).lower()
+            # get only the alphanumeric characters
+            objstr = "".join(e for e in objstr if e.isalnum())
+            # generate hash
+            m.update(f"{pepper}:{objstr}".encode())
+            # return first 32 characters of the 64 character hash hex
+            return m.hexdigest()[:32]
+    return None
+
+
+def persist_answers_log(
+    is_spl_match=None, submission_reference=None, submission_details=[], nhs_sub=None
```

Comment:
spl match needs removing as it is no longer done on the front end

Reply:
Done